qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -74,27 +75,22 @@ def mul_monomial(self, indices):
         """
         length = len(indices)
         assert length < 4, "no term!"
-        assert True not in [x < 0 or x >= self.n_vars for x in indices], \
-            "indices out of bounds!"
-        assert False not in [indices[i] < indices[i+1]
-                             for i in range(length-1)], \
-            "indices non-increasing!"
+        assert ((np.array(indices) < 0).all() is not False) and \
```

Comment:
`(np.array(indices) < 0).all() is not False` seems wrong to me if we want to verify we have no negative indices; it can be false if not all the indices are <0, i.e. only some are negative.

Also, why convert twice to an array?

I suggest:
```
indices_arr = np.array(indices)
assert((indices_arr >= 0).all() and (indices_arr < self.n_vars).all(), "indices out of bounds!")
```